test/routing/route_test.cc

```diff
@@ -726,6 +729,582 @@ TEST_F(RouteLaneSRelationTest, EvaluateComputeLaneSRangeRelationSameLaneSRanges)
+                                           kEndRoadPositionsPhaseThree, kLaneSRangesPhaseThree, road_network_.get());
+    phase_four_ = std::make_unique<Phase>(kPhaseFourIndex, kLaneSRangeTolerance, kStartRoadPositionsPhaseFour,
+                                          kEndRoadPositionsPhaseFour, kLaneSRangesPhaseFour, road_network_.get());
+    EXPECT_CALL(*road_geometry_ptr_, do_linear_tolerance()).WillRepeatedly(Return(kLinearTolerance));
+  }
+
+ protected:
+  LaneMock lane_d_;
+  LaneMock lane_e_;
+  LaneMock lane_f_;
+  LaneMock lane_g_;
+  SegmentMock segment_two_;
+  SegmentMock segment_three_;
+  SegmentMock segment_four_;
+  std::unique_ptr<Phase> phase_two_;
+  std::unique_ptr<Phase> phase_three_;
+  std::unique_ptr<Phase> phase_four_;
+};
+
+TEST_F(RouteComputeLaneSRouteCaseFTest, ComputeLaneSRoute) {
+  const api::RoadPosition kRoadPositionA{&lane_a_, kQueryLanePosition};
+  const api::RoadPosition kRoadPositionB{&lane_b_, kQueryLanePosition};
+  const api::RoadPosition kRoadPositionC{&lane_c_, kQueryLanePosition};
+  const api::RoadPosition kRoadPositionD{&lane_d_, kQueryLanePosition};
+  const api::RoadPosition kRoadPositionE{&lane_e_, kQueryLanePosition};
+  const api::RoadPosition kRoadPositionF{&lane_f_, kQueryLanePosition};
+  const api::RoadPosition kRoadPositionG{&lane_g_, kQueryLanePosition};
+  const api::LaneSRoute kExpectedLaneSRouteFromA(
+      {kLaneSRangeA, kLaneSRangeB, kLaneSRangeC, kLaneSRangeD, kLaneSRangeE, kLaneSRangeF, kLaneSRangeG});
+  const api::LaneSRoute kExpectedLaneSRouteFromB(
+      {kLaneSRangeB, kLaneSRangeC, kLaneSRangeD, kLaneSRangeE, kLaneSRangeF, kLaneSRangeG});
+  const api::LaneSRoute kExpectedLaneSRouteFromC(
+      {kLaneSRangeC, kLaneSRangeD, kLaneSRangeE, kLaneSRangeF, kLaneSRangeG});
+  const api::LaneSRoute kExpectedLaneSRouteFromD({kLaneSRangeD, kLaneSRangeE, kLaneSRangeF, kLaneSRangeG});
+  const api::LaneSRoute kExpectedLaneSRouteFromE({kLaneSRangeE, kLaneSRangeF, kLaneSRangeG});
+  const api::LaneSRoute kExpectedLaneSRouteFromF({kLaneSRangeF, kLaneSRangeG});
+  const api::LaneSRoute kExpectedLaneSRouteFromG({kLaneSRangeG});
+  const Route dut({*phase_zero_, *phase_one_, *phase_two_, *phase_three_, *phase_four_}, road_network_.get());
+
+  EXPECT_TRUE(AssertCompare(IsEqual(kExpectedLaneSRouteFromA, dut.ComputeLaneSRoute(kRoadPositionA))));
+  EXPECT_TRUE(AssertCompare(IsEqual(kExpectedLaneSRouteFromB, dut.ComputeLaneSRoute(kRoadPositionB))));
+  EXPECT_TRUE(AssertCompare(IsEqual(kExpectedLaneSRouteFromC, dut.ComputeLaneSRoute(kRoadPositionC))));
+  EXPECT_TRUE(AssertCompare(IsEqual(kExpectedLaneSRouteFromD, dut.ComputeLaneSRoute(kRoadPositionD))));
+  EXPECT_TRUE(AssertCompare(IsEqual(kExpectedLaneSRouteFromE, dut.ComputeLaneSRoute(kRoadPositionE))));
+  EXPECT_TRUE(AssertCompare(IsEqual(kExpectedLaneSRouteFromF, dut.ComputeLaneSRoute(kRoadPositionF))));
+  EXPECT_TRUE(AssertCompare(IsEqual(kExpectedLaneSRouteFromG, dut.ComputeLaneSRoute(kRoadPositionG))));
+}
```

Comment:
Can we add additional test cases that are variations of Case F?

```
// Case G:
// *----A>----x----B>----*          *----F>----x----G>----*
//            *----C>----x----D>----x----E>----*


// Case H:
// *----A>----x----B>----*
//            *----C>----x----D>----x----E>----*
//                                  *----F>----x----G>----*

// Case I:
//                                  *----A>----x----B>----*
//            *----C>----x----D>----x----E>----*
// *----F>----x----G>----*
```

I'm also wondering if we could test multiple lateral hops in the same phase:

```
// *----A>----x----B>----*          *----H>----x----I>----*
//            *----C>----*          *----G>----*
//            *----D>----x----E>----x----F>----*
```

Reply:
Maybe also a more complex scenario with multi-option branches?
```
// *----A>----x----B>----*          *----H>----x----I>----*
//            *----C>----*          *----G>----*
//            *----D>----x----E>----x----F>----*
// *----J>----x----K>----*          *----L>----x----M>----*
//            *----N>----*          *----O>----*
```

Reply:
I don't think the last one is a valid Route. A and J, as well as I and M should be in the same Phase but they don't seem to be adjacently connected. It is weird to me that we would have geometries that are adjancently connected but would follow to a completely different width description such that you end up having lanes in between with no common branchpoints. I can work on the others.

Reply:
Sorry, I am not confident regarding what is valid versus not. Definitely only test valid routes! I was just trying to push the complexities of these routes to the max. Is a route with a doughnut shape in it valid?

```
//            *----A>----x----B>----x----C>----*          *----D>----x----E>----*
//            *----C>----*          *----C>----*          *----G>----*
// *----F>----x----G>----*          *----H>----x----I>----x----J>----*
//            *----K>----x----L>----x----M>----*
```

Reply:
I've omitted case G, H and I above because they present variations covered. I think, in terms of the nomenclature selected in c7b8e5b, we have a pretty good coverage of route topologies.

Regarding the doughnut route in https://github.com/maliput/maliput/pull/557#discussion_r1416039278, we don't support such topology.